orchagent/lagids.lua

```diff
@@ -0,0 +1,90 @@
+-- KEYS - None
+-- ARGV[1] - operation (add/del/get)
+-- ARGV[2] - lag name
+-- ARGV[3] - current lag id (for "add" operation only)
+
+-- return lagid if success for "add"/"del"
+-- return 0 if lag not exists for "del"
+-- return -1 if lag table full for "add"
+-- return -2 if log does exist for "get"
+-- return -3 if invalid operation
+
+local op = ARGV[1]
+local pcname = ARGV[2]
+
+local lagid_start = tonumber(redis.call("get", "SYSTEM_LAG_ID_START"))
+local lagid_end = tonumber(redis.call("get", "SYSTEM_LAG_ID_END"))
+
+if op == "add" then
+
+    local plagid = tonumber(ARGV[3])
+
+    local dblagid = redis.call("hget", "SYSTEM_LAG_ID_TABLE", pcname)
+
+    if dblagid then
+        dblagid = tonumber(dblagid)
+        if plagid == 0 then
+            -- no lagid propsed. Return the existing lagid
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(dblagid))
+            return dblagid
+        end
+    end
+
+    -- lagid allocation request with a lagid proposal
+    if plagid >= lagid_start and plagid <= lagid_end then
+        if plagid == dblagid then
+            -- proposed lagid is same as the lagid in database
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(plagid))
```

Comment:
We could actually skip this redis.call, as the SET already has this id ?

Reply:
Yes. "sadd" can be skipped here. I'll fix this. 
The "sadd" in line#28 can also be skipped. I'll fix this also.

Reply:
Fixed